src/main/java/io/pravega/connectors/flink/sink/PravegaSinkBuilder.java

```diff
@@ -81,20 +82,42 @@ public PravegaSinkBuilder<T> forStream(final Stream stream) {
     }
 
     /**
-     * Sets the writer mode to provide at-least-once or exactly-once guarantees.
+     * Sets the delivery guarantee to provide at-least-once or exactly-once guarantees.
+     *
+     * @param deliveryGuarantee The delivery guarantee of {@code NONE}, {@code AT_LEAST_ONCE}, or {@code EXACTLY_ONCE}.
+     * @return A builder to configure and create a sink.
+     */
+    public PravegaSinkBuilder<T> withDeliveryGuarantee(DeliveryGuarantee deliveryGuarantee) {
+        this.deliveryGuarantee = deliveryGuarantee;
+        return this;
+    }
+
+    /**
+     * Sets the delivery guarantee from writer mode to provide at-least-once or exactly-once guarantees.
+     * This method is to keep the compatibility with the old {@code PravegaWriterMode} enum that will be deprecated soon.
      *
      * @param writerMode The writer mode of {@code BEST_EFFORT}, {@code ATLEAST_ONCE}, or {@code EXACTLY_ONCE}.
      * @return A builder to configure and create a sink.
```

Comment:
Add a deprecation notice here, suggesting user to switch to `withDeliveryGuarantee` method